testng-core/src/main/java/org/testng/internal/invokers/MethodInvocationHelper.java

```diff
@@ -390,13 +393,16 @@ private static void invokeWithTimeoutWithNewExecutor(
     boolean finished = exec.awaitTermination(realTimeOut, TimeUnit.MILLISECONDS);
 
     if (!finished) {
-      exec.shutdownNow();
       ThreadTimeoutException exception =
           new ThreadTimeoutException(
               "Method "
                   + tm.getQualifiedName()
                   + "() didn't finish within the time-out "
                   + realTimeOut);
+      StackTraceElement[] realStackTrace = getRunningMethodStackTrace(exec);
+      if (realStackTrace != null) exception.setStackTrace(realStackTrace);
```

Comment:
Same case here as well. Please include the if condition true part within `{}` for readability sake.